### ISSUE TITLE
Weekly view doesn't display correct week when setting "Start week on Sunday" enabled

### ISSUE THREAD

**Issue by maze88** [SimpleMobileTools/Simple-Calendar#1894]
Recreation:
The bug occurred for me today (_Sunday_, 2022-12-4), where my calendar's weekly view was still on the previous week (2022-11-27 through 2022-12-3).

Version:
6.20.3 Pro

Notes:
- Not resolved by force-stopping, and restarting the app.

**Comment by tibbi** [comment]
you mean not even after restarting the app?

**Comment by maze88** [comment]
> you mean not even after restarting the app?

Nope.
(I updated this detail in original post)

**Comment by maze88** [comment]
Update:
At some point in the afternoon (I noticed at about 17:30) that the issue no longer existed... you may need deeper investigation on how to recreate...

**Comment by tibbi** [comment]
well, guess it was related to being a Sunday yesterday, while it is a Monday already

**Comment by maze88** [comment]
But yesterday [afternoon] was still totally Sunday.
I wouldn't even suspect a timezone issue, since being in +0200 wouldn't explain the issue either.

**Comment by maze88** [comment]
Re-confirming this issue still occurs (I am now on version 6.21.3 Pro).

**Comment by WessellUrdata** [comment]
Unfortunately, it is _somewhat_ related to timezone.

I tested several timezones by changing the time and timezone on my phone and then killing the app and relaunching it/clicking the Gtt button (Go to today) on the top bar to get the displayed week.

App version: v6.21.4 F-Droid version

The results are as follows:

**UTC+8 HKT, Hong Kong, 2023-01-29: (does NOT observe DST)**

00:00 : Gtt points to wk4 (incorrect)
20:59 : Gtt points to wk4 (incorrect)
21:00 : Gtt points to wk5 (correct)
23:59 : Gtt points to wk5 (correct)

20:59/21:00 is the cutoff.

---

**UTC+2 IST, Israel, 2023-01-29: (observes DST, but currently not in effect)**


14:59 : Gtt points to wk4 (incorrect)
15:00 : Gtt points to wk5 (correct)

14:59/15:00 is the cutoff.

---

**UTC/GMT, London, 2023-01-29: (observes DST, but currently not in effect)**

12:59 : Gtt points to wk4 (incorrect)
13:00 : Gtt points to wk5 (correct)


12:59/13:00 is the cutoff.

---

**UTC-5 EST, New York, 2023-01-29: (observes DST, but currently not in effect)**
Weekly view - Start day at 08:00

07:59 : Gtt points to wk4 (incorrect)
08:00 : Gtt points to wk5 (correct)

07:59/08:00 is the cutoff.

---

The cutoff time at which the weekly view becomes correct is correlated with the difference in timezone/UTC offset.

Other observations that I found:

- Changing **Weekly view - Start day at** [time] does not affect the cutoff time. I tested at 08:00 and 07:00 and they yield the same result.
- Australia is an oddball. Here are the results I got:

  **UTC+11 AEDT, Sydney, 2023-01-29: (DST in effect)**
  19:59/20:00 is the cutoff.
  
  **UTC+10 AEST, Sydney, 2023-04-09: (DST NOT in effect)**
  18:59/19:00 is the cutoff.
  
  **UTC+10 AEST, Brisbane, 2023-01-29: (does NOT observe DST)**
  18:59/19:00 is the cutoff.
  
  **UTC+10 AEST, Brisbane, 2023-04-09: (does NOT observe DST)**
  18:59/19:00 is the cutoff.
  
  While it does compensate for DST, the cutoff time do not fall in line with the the first 4 timezones I tested with regards to the UTC offset.

**Comment by WessellUrdata** [comment]
While writing a fix for this bug, I've discovered another problem:

Even with "Start week with the current day" enabled, regardless of the day of week, it might not show the current day as the first day.

I've found this function to be the culprit:
https://github.com/SimpleMobileTools/Simple-Calendar/blob/d268c2910e04e17ae66490cd11a56085638bbe38/app/src/main/kotlin/com/simplemobiletools/calendar/pro/extensions/Context.kt#L767-L781

One of the problems for its failing is because of Joda-Time treating Monday as 1st day of the week (Day of Week 1), and Sunday as 7th day of the week.

But for "Start week with the current day", since it returns the UTC's 1st day of the week, if you live somewhere that doesn't use UTC timezone (for example, me living in UTC +8 when it's 0100), there may be a discrepency.
